engine.py

```diff
@@ -912,3 +912,58 @@ def __setup_favorite_dirs(self):
                                 type=(nuke.IMAGE | nuke.SCRIPT | nuke.GEO),
                                 icon=icon_path,
                                 tooltip=path)
+
+    #####################################################################################
+    # Script and session related methods
+
+    def get_session_path(self):
+        """
+        Returns the path to the current file if it resides on
+        disk. If unsaved, it returns None.
+
+        :return: Path to the current scene if file is saved, else returns
+                 and empty string.
+        """
+        # retrieve the name of the nuke file. In case of a saved file, the
+        # absolute path of the file is returned, for an unsaved file,
+        # an empty string is returned.
+        return nuke.root().knob("name").value()
+
+    def get_session_dependencies(self):
+        """
+         Returns the list of file or folder dependencies for the current
+         session.
+
+         :return: A list of file or folder paths that the current
+                  session depends on.
+         """
+        dependencies = []
+
+        # we shall query for file dependencies for the file frame range
+        start_frame = int(nuke.root().knob("first_frame").value())
+        end_frame = int(nuke.root().knob("last_frame").value())
+
+        # create a list of enabled write node in the session
+        enabled_write_nodes = []
+        write_nodes = [node for node in nuke.allNodes()
+                       if node.Class() == "Write"]
+        if write_nodes:
+            enabled_write_nodes = [node for node in write_nodes
```

Comment:
again, i think we want the dependencies to reflect the set of inputs into the scene, regardless of how those are being used. Imagine an application where we want to perform a remote transfer of work from one artist to another - in that case, it probably makes sense to transfer the "disabled" content.